docs/src/main/asciidoc/datastore.adoc

```diff
@@ -985,6 +985,37 @@ If a method's return type is `org.springframework.data.domain.Page`, the returne
 NOTE: Methods that return `Page` execute an additional query to compute total number of pages.
 Methods that return `Slice`, on the other hand, don't execute any additional queries and therefore are much more efficient.
 
+==== Empty result handling in repository methods
+Java `java.util.Optional` can be used to indicate the potential absence of a return value.
+
+Alternatively, query methods can return the result without a wrapper.
+In that case the absence of a query result is indicated by returning `null`.
+Repository methods returning collections are guaranteed never to return `null` but rather the corresponding empty collection.
+
+===== Nullability Annotations
+You can express nullability constraints for repository methods by using https://docs.spring.io/spring/docs/5.2.5.RELEASE/spring-framework-reference/core.html#null-safety[Spring Framework’s nullability annotations].
```

Comment:
I took another look at this documentation and realized that this is part of core Spring and is not specific to Spring Data at all. Hence, I don't think we need to make a big deal out of it here. I think it's enough to mention that if you want null checks use null-safety annotations and link to them.
Also, please use the non-version-specific link: https://docs.spring.io/spring/docs/current/spring-framework-reference/core.html#null-safety